src/infohash.rs

```diff
@@ -56,6 +56,12 @@ impl Infohash {
   }
 }
 
+impl Into<[u8; 20]> for Infohash {
```

Comment:
I'm unsure I like this conversion, I feel like I'm not sure I want `.into()` conversions of `[u8; 20]` to Infohashes in the codebase, but I'm honestly not sure it's totally rational. It feels too general. It think maybe the conversion should be `Infohash::new(sha1hash)`

Reply:
I think the advice these days is to do `Impl From<Infohash> for [u8; 20]` and get the `into()` for free. That's possibly why you \~feel\~ weird?

Reply:
I misread it as being a conversion from `[u8; 20]` to an `Infohash`. So nevermind my original objection. But yes, this should be `impl From<Infohash>`.